mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/source/SourceCallbackContext.java

```diff
@@ -112,4 +115,16 @@
    * @return The {@link SourceCallback} that generated {@code this} context
    */
   <T, A> SourceCallback<T, A> getSourceCallback();
+
+  /**
+   * Indicates that an {@link ExtensionNotification} should be fired with the desired information when the runtime takes the
```

Comment:
`when the runtime takes the source result to process it.` what does this mean? I should be able to fire a notification whenever i want, right?

Reply:
This regards the `Event` problem. You don't have one at this point and firing a custom notification not associated to one would not allow correlating that notification to later processing. So the actual firing will be done once the `Event` is created.